test-result-summary-client/src/Dashboard/Widgets/Graph/ODM.jsx

```diff
@@ -80,14 +80,13 @@ export default class ODM extends Component {
 
         // combine results having the same JDK build date
         results.forEach(( t, i ) => {
-            if ( t.buildResult !== "SUCCESS" ) return;
-
+            if ( t.buildResult !== "SUCCESS" || ! Array.isArray(t.aggregateInfo) || t.aggregateInfo.length < 1 || ! t.aggregateInfo[0].jdkDate || ! Array.isArray(t.aggregateInfo[0].metrics) || t.aggregateInfo[0].metrics.length < 1 || ! t.aggregateInfo[0].metrics[0]["value"] || Object.keys(t.aggregateInfo[0].metrics[0]["value"]).length === 0) return;

```

Comment:
Some concerns regarding these checks:

1) When do we store an empty array `t.aggregateInfo.length < 1`? I believe there isn't any use for doing so.
2) Same issue as 1 for `t.aggregateInfo[0].metrics.length < 1`

4)

Reply:
Used validAggregateInfo to identify all the checks here.